packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -17,6 +17,13 @@ const MY_STACK_CACHE = Symbol.for('@aws-cdk/core.Stack.myStack');
 
 const VALID_STACK_NAME_REGEX = /^[A-Za-z][A-Za-z0-9-]*$/;
 
+/**
+ * The well-known name for the docker image asset ECR repository. All docker
+ * image assets will be pushed into this repository with an image tag based on
+ * the source hash.
+ */
+const ASSETS_ECR_REPOSITORY_NAME = "aws-cdk/assets";
```

Comment:
I'd prefer if you didn't do this yet. You can remove ECR repository cleanup while still keeping the mechanism of passing in the image via parameters the same. There will be project step of replacing parameters w/ hardcoded asset locations in the CI/CD project later on. Combining the two changes muddles the water, imo.

Reply:
Is this comment about using a single repo for all images or about passing the repo name from the CLI through a parameter (or both)?

If the app knows the repository name and image tag during synth (whether it's a single repo or multiple repos), there is no need to pass the repo name through a parameter. 

Do you mean that maybe users are leveraging the parameter through some other means (i.e. bespoke CI/CD setups)? One of the reasons I wanted to front load this change is to uncover all these undocumented [ab]usages of the legacy setup, to reduce the risks around our CI/CD implementation.

Happy to change either of these if you feel this will complicate the implementation of CI/CD in any way of course.

Reply:
Okay, as discussed, we decided to go on with this change.